src/TestFramework/PhpUnit/Coverage/IndexXmlCoverageParser.php

```diff
@@ -0,0 +1,293 @@
+
+        $fileName = $fileNode->getAttribute('name');
+        $relativeFilePath = $fileNode->getAttribute('path');
+
+        if ($relativeFilePath !== '') {
+            // The relative path is not present for old versions of PHPUnit. As a result we parse
+            // the relative path from the source file path and the XML coverage file
+            $relativeFilePath = str_replace(
+                sprintf('%s.xml', $fileName),
+                '',
+                $relativeCoverageFilePath
+            );
+        }
+
+        $path = $projectSource . '/' . ltrim($relativeFilePath, '/') . '/' . $fileName;
+        $realPath = native_realpath($path);
+
+        if ($realPath === false) {
+            throw CoverageDoesNotExistException::forFileAtPath($fileName, $path);
+        }
+
+        return $realPath;
+    }
+
+    /**
+     * @param DOMNodeList|DOMElement[] $coveredLineNodes
+     *
+     * @return array<int, array<int, CoverageLineData>>
+     */
+    private static function collectCoveredLinesData(DOMNodeList $coveredLineNodes): array
+    {
+        $data = [];
+
+        foreach ($coveredLineNodes as $lineNode) {
+            $lineNumber = $lineNode->getAttribute('nr');
+
+            Assert::integerish($lineNumber);
+
+            $lineNumber = (int) $lineNumber;
+
+            /** @var DOMNodeList|DOMElement[] $coveredNodes */
+            $coveredNodes = $lineNode->childNodes;
+
+            foreach ($coveredNodes as $coveredNode) {
+                if ($coveredNode->nodeName !== 'covered') {
+                    continue;
+                }
+
+                $data[$lineNumber][] = CoverageLineData::withTestMethod(
+                    $coveredNode->getAttribute('by')
+                );
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param DOMNodeList|DOMElement[] $methodsCoverageNodes
+     *
+     * @return MethodLocationData[]
+     */
+    private static function collectMethodsCoverageData(DOMNodeList $methodsCoverageNodes): array
+    {
+        $methodsCoverage = [];
+
+        foreach ($methodsCoverageNodes as $methodsCoverageNode) {
+            $methodName = $methodsCoverageNode->getAttribute('name');
+
+            $start = $methodsCoverageNode->getAttribute('start');
+            $end = $methodsCoverageNode->getAttribute('end');
+
+            Assert::integerish($start);
+            Assert::integerish($end);
+
+            $methodsCoverage[$methodName] = new MethodLocationData(
+                (int) $start,
+                (int) $end
+            );
+        }
+
+        return $methodsCoverage;
+    }
+
+    private static function getProjectSource(DOMXPath $xPath): string
+    {
+        // PHPUnit >= 6
+        $sourceNodes = self::safeQuery($xPath, '//project/@source');
+
+        if ($sourceNodes->length > 0) {
+            return $sourceNodes[0]->nodeValue;
+        }
+
+        // PHPUnit < 6
+        return self::safeQuery($xPath, '//project/@name')[0]->nodeValue;
```

Comment:
Can't see when this happens either even with PHPUnit 5... Could you point it out so we could cover it?

Reply:
You can reproduce it with this repo https://github.com/maks-rafalko/tactician-domain-events

Just checked it

```xml
<?xml version="1.0"?>
<phpunit xmlns="http://schema.phpunit.de/coverage/1.0">
  <project name="/Users/maksrafalko/sites/tactician-domain-events/src">
```

Reply:
👍 Adding a test for it then